### ISSUE TITLE
Image installer does not resolve prereleases for compatible specifiers

### ISSUE THREAD

**Issue by alnlarsen** [opentap/opentap#1866]
This is closely related to #1863

Consider these two scenarios:
1. `tap image install 'Basic Mixins:^0' -r ~/Downloads`
2. `tap image install 'Basic Mixins:^0' -r https://packages.opentap.io`

The first scenario succeeds (if basic mixins is available in local caches)
The second scenario fails because no release version is available, and the resolver does not fetch prereleases in this scenario.

Actual behavior: Resolution fails
Expected: It should resolve 0.2.0-beta.1 (as it does when using local caches)

When a resolution fails for a *compatible* version specifier with no specified prerelease, the resolver must fetch all prereleases and try again.

**Comment by tudor9507** [comment]
Verified in Open TAP 9.27.1 rc1. The issue cannot be reproduce anymore, basic mixins is installed from remote repo but an older version of open tap is installed. 
Is this the expected behavior?

![image](https://github.com/user-attachments/assets/21e400e6-71cb-49bf-bbdf-2bf1429c4a9e)
@alnlarsen

**Comment by alnlarsen** [comment]
@tudor9507 yes, this is expected because the resolved version of Basic Mixins depends on OpenTAP 9.25.1

**Comment by tudor9507** [comment]
> @tudor9507 yes, this is expected because the resolved version of Basic Mixins depends on OpenTAP 9.25.1

marked as qa approved based on comment
